uint32 obj_id
VoxelGridBool voxel
